src/jslogger.js

```diff
@@ -0,0 +1,320 @@
+;(function() {
+    'use strict';
+
+    function jslogger(req) {
+
+        this.ajax      = ajax;
+        this.getJSON   = getJSON;
+        this.get       = get;
+        this.post      = post;
+        this.exception = exception;
+        this.info      = info;
+        this.error     = error;
+        this.debug     = debug;
+        this.log       = log;
+        this.warn      = warn;
+        this.message   = msg;
+        this.bind      = bind;
+
+        var queue           = 0,
+            count           = 0,
+            url             = req.url,
+            timeInterval    = req.time_ms ? req.time_ms : 5000,
+            maxLogs         = req.max_logs ? req.max_logs : 1000,
+            ajaxCall        = ajaxCall,
+            common          = common,
+            appender        = appender,
+            user            = null,
+            user_detail     = null,
+            isAjaxCompleted = true;
+
+        function bind(req){
+
+            user = req.user;
+            delete req.user;
+            user_detail = req;
+        }
+
+        // Get data from all kinds of messages that are being logged and form a structed log.
+        function common(options) {
+
+            var time = new Date();
+            // TODO use window.location.host
+            var source = window.location.host
+            var data = {
+                UUID      : uuidv1(),
+                timestamp : time.getTime(),
+                message   : options.message,
+                level     : options.type,
+                url       : options.url,
+                host_url  : source,
+                misc      : options.misc
+            }
+
+            if(user != null) {
+                data.user = user;
+            }
+            if(user_detail != null) {
+                for(var key in user_detail) {
+                    data[key] = user_detail[key];
+                }
+            }
+
+            appender(data);
```

Comment:
As mentioned earlier, write variable declarations and function definitions first before using them, here appender is not declared before. Its a good practice